src/Lean/Parser/Tactic.lean

```diff
@@ -66,13 +66,55 @@ doing a pattern match. This is equivalent to `fun` with match arms in term mode.
 @[builtin_tactic_parser] def introMatch := leading_parser
   nonReservedSymbol "intro" >> matchAlts
 
-/-- `decide` will attempt to prove a goal of type `p` by synthesizing an instance
-of `Decidable p` and then evaluating it to `isTrue ..`. Because this uses kernel
-computation to evaluate the term, it may not work in the presence of definitions
-by well founded recursion, since this requires reducing proofs.
-```
+/--
+`decide` attempts to prove a goal of type `p` by synthesizing an instance
+of `Decidable p` and then reducing that instance. If it reduces to `isTrue h`,
+then `h` is used to close the goal.
+
+Limitations:
+- The goal is not allowed to have local variables or metavariables.
+  If there are local variables, you can try to `revert` them first.
```

Comment:
Slight pedantic wording thing:
```suggestion
- The goal is not allowed to contain local variables or metavariables.
  If there are local variables, you can try to `revert` them first.
```

Do we expect there to be enough understanding of `revert` in the audience for this? Or should it be a bit more explanatory, like:
```suggestion
- The goal is not allowed to refer to local variables or contain metavariables.
  The `revert` tactic can transform local variable references into bindings in the
  goal, which may allow `decide` to succeed.
```